### ISSUE TITLE
[ML] Better handling for jobs which will may fail

### ISSUE THREAD

**Issue by sophiec20** [elastic/elasticsearch#39704]
Found in `8.0.0-SNAPSHOT`, expected in other versions.

Usability issue.

In Dev Tools, if I create a job with a duplicate detector, the call to open the job will succeed. However the job then subsequently becomes "failed". It is difficult for the end user to know anything is wrong unless they check the job state or try to start the datafeed (at which point the datafeed fails to start because the job is not open).

Note. The UI does not allow you to create a job with a duplicate detector, so this issue concerns Dev Tools and API.

```
PUT _ml/anomaly_detectors/fq2
{
  "analysis_config": {
    "bucket_span": "15m",
    "detectors": [
      { "function": "mean", "field_name": "responsetime" },
      { "function": "mean", "field_name": "responsetime" }
    ]
  },
  "data_description": {
    "time_field": "@timestamp"
  }
}
// succeeds


PUT _ml/datafeeds/datafeed-fq2
{
  "job_id": "fq2",
  "indices": [ "farequote-*" ]
}
// succeeds

POST _ml/anomaly_detectors/fq2/_open
// succeeds 
// { "opened" : true }

POST _ml/datafeeds/datafeed-fq2/_start
// fails
// reason": "cannot start datafeed [datafeed-fq2] because job [fq2] is failed"

```

Can validation checks be implemented when the job is `PUT`?
Or, can the call to `_open` fail?
And are there other similar reasons for the job to fail that could be caught earlier with more user feedback?

**Comment by droberts195** [comment]
This is a duplicate of #30070

**Comment by davidkyle** [comment]
The duplicate detector bug is known issue #30070. It couldn't be addressed when the config was stored in clusterstate as it meant changing the validation which could potentially error parsing of the clusterstate but it can be fixed now config has moved to an index.

The error occurs when autodetect is starting up. When a job is opened autodetect is launched and the code assumes it is happily working, there is no ability to wait for autodetect to report that it has parsed the config and started without error. Also this is very difficult to do on an open job request as the request will hit the master node but the job can start (and possibly fail) on a different node. The persistent tasks framework is used to detect that the job has started (or failed) on another node then the master node responds to the original request. There should be a more visible way to surface these errors and that is a little easier after #38029

**Comment by droberts195** [comment]
An attempt was made to fix this when we were in the private X-Pack repo (see the reverse link), but it was aborted because:

> At the moment of creating this PR I realised it cannot be merged due to BWC issues. If someone has a job defined with duplicate detectors then after upgrade parsing the analysis config will throw an exception blocking the node.

This problem will not be quite so severe for job configs stored in indices, so from 8.0 we probably should change the validation in the Java to be the same as that in the C++.  It's extremely unlikely that any user will be affected, and if they are it will only cripple ML, not the whole cluster.

**Comment by droberts195** [comment]
The other thing we could do starting in any release is to introduce a separate validation method that is not called during parsing but can be called _only_ by the put job action and validate job action.  This cannot create any BWC problems with existing jobs, but will prevent new jobs being added that fail the tighter validation.  If a user already has a job that is impossible to run without it failing due to C++ validation then they'll still have this job until they delete it.

**Comment by droberts195** [comment]
Fixes for the remaining problems listed in #29843 should be incorporated into any changes made for this issue.
